glean-core/android/src/test/java/mozilla/telemetry/glean/scheduler/MetricsPingSchedulerTest.kt

```diff
@@ -595,6 +595,86 @@ class MetricsPingSchedulerTest {
         }
     }
 
+    @Test
+    fun `Glean must preserve lifetime application metrics across runs`() {
+        // This test requires to use the glean instance (it's more an integration
+        // test than a unit test).
+        val context = getContextWithMockedInfo()
+
+        // Reset Glean and do not start it right away.
+        Glean.testDestroyGleanHandle()
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.setTaskQueueing(true)
+
+        // Let's create a fake time the metrics ping was sent: this is required for
+        // us to not send a 'metrics' ping the first time we init glean.
+        val fakeNowDoNotSend = Calendar.getInstance()
+        fakeNowDoNotSend.clear()
+        fakeNowDoNotSend.set(2015, 6, 11, 4, 0, 0)
+        SystemClock.setCurrentTimeMillis(fakeNowDoNotSend.timeInMillis)
+
+        // Create a fake instance of the metrics ping scheduler just to set the last
+        // collection time.
+        val fakeMpsSetter = spy(MetricsPingScheduler(context))
+        fakeMpsSetter.updateSentDate(getISOTimeString(fakeNowDoNotSend, truncateTo = TimeUnit.Day))
+
+        // Create a metric with lifetime: application and set it. Put
+        // it in the metrics ping so that we can easily trigger it for
+        // the purpose of this test.
+        val testMetric = StringMetricType(
+            disabled = false,
+            category = "telemetry",
+            lifetime = Lifetime.Application,
+            name = "test_applifetime_metric",
+            sendInPings = listOf("metrics")
+        )
+        val expectedString = "I-will-survive!"
```

Comment:
![](https://media1.giphy.com/media/qDIeQmPWXbo3e/giphy.gif)

Reply:
Haha, you got it ;-)